crates/router/src/core/payments/operations/payment_session.rs

```diff
@@ -299,84 +309,88 @@ where
-                            ) {
-                                let connector_and_wallet = (
+                            .unwrap_or_default()
+                            .into_iter()
+                            .filter(|payment_method_type| {
+                                let is_invoke_sdk_client = matches!(
+                                    payment_method_type.payment_experience,
+                                    Some(api_models::enums::PaymentExperience::InvokeSdkClient)
+                                );
+
+                                // If session token is requested for the payment method type,
+                                // filter it out
+                                // if not, then create all sessions tokens
+                                let is_sent_in_request = requested_payment_method_types
+                                    .contains(&payment_method_type.payment_method_type)
+                                    || requested_payment_method_types.is_empty();
+
+                                is_invoke_sdk_client && is_sent_in_request
+                            })
+                            .map(|payment_method_type| {
+                                (
                                     connector_account.connector_name.to_owned(),
                                     payment_method_type.payment_method_type,
-                                );
-                                connector_and_supporting_payment_method_type
-                                    .push(connector_and_wallet);
-                            }
-                        }
-                    }
-                    Err(parsing_error) => {
-                        logger::debug!(session_token_parsing_error=?parsing_error);
-                    }
+                                    connector_account.business_sub_label.to_owned(),
+                                )
+                            })
+                            .collect::<Vec<_>>()
+                    })
+                    .collect::<Vec<_>>();
+                connector_and_supporting_payment_method_type.extend(res);
+            });
+
+        let mut session_connector_data =
+            Vec::with_capacity(connector_and_supporting_payment_method_type.len());
+
+        for (connector, payment_method_type, business_sub_label) in
+            connector_and_supporting_payment_method_type
+        {
+            match api::ConnectorData::get_connector_by_name(
+                connectors,
+                &connector,
+                api::GetToken::from(payment_method_type),
+            ) {
+                Ok(connector_data) => session_connector_data.push(api::SessionConnectorData {
+                    payment_method_type,
+                    connector: connector_data,
+                    business_sub_label,
+                }),
+                Err(error) => {
+                    logger::error!(session_token_error=?error)
                 }
```

Comment:
Here too, shouldn't we throw an error?

Reply:
No we shouldn't, for the same reason as specified before. I'm thinking of collecting all the errors and then return them in a separate field in the response.